src/fam-api/fam.cpp

```diff
@@ -865,6 +889,84 @@ int fam::Impl_::validate_item(Fam_Descriptor *descriptor) {
     return 0;
 }
 
+/*
+ * find_config_file - Look for configuration file in user's home directory
+ * or in path specified by OPENFAM_ROOT environment variable or in /opt/OpenFAM.
+ * On success, input parameter config file is updated with correct path.
+ * @throws: InvalidOption_Exception if configuration file is not found in any of
+ * paths mwntioned above.
+ */
+void fam::Impl_::find_config_file(char *filetype, char *config_file) {
+    struct stat buffer;
+
+    // Look for config file in user home directory
+    char *homedir = getenv("HOME");
+    if (homedir != NULL) {
+        sprintf(config_file, "%s/%s/%s", homedir, "config", filetype);
+    }
+
+    if (stat(config_file, &buffer) == 0) {
+        return;
+    }
+
+    // Look for config file in OPENFAM_ROOT
+    char *openfam_root = getenv("OPENFAM_ROOT");
+
+    if (openfam_root != NULL) {
+        sprintf(config_file, "%s%s/%s", openfam_root, "config", filetype);
+    }
+    if (stat(config_file, &buffer) == 0) {
+        return;
+    }
+
+    // Look for config file in /opt
+    sprintf(config_file, "%s/%s/%s", "/opt/OpenFAM/", "config", filetype);
+    if (stat(config_file, &buffer) == 0) {
+        return;
+    }
+
+    std::ostringstream message;
+    message << "Fam Config: Configuration file not found.";
+    throw Fam_InvalidOption_Exception(message.str().c_str());
```

Comment:
Should we be throwing exception, if config file is not found ? @faizan-barmawer . I guess we should use the config default from code.